### ISSUE TITLE
Removing an item from a menu in the Nav block list view causes a focus loss. 

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#48017]
Identified during a11y review https://github.com/WordPress/gutenberg/issues/46939#issuecomment-1427668766, when removing an item using the Nav block list view focus is completely lost. It should be returned to the previous node or at least somewhere within the list view.

**Comment by draganescu** [comment]
Investigating this I learned that:

- removing any block moves focus to the editor `iframe`
- removing a submenu items, both in canvas and in the inspector, moves focus to document's `body`
- removing a menu item via the canvas moves focus to the editor `iframe`
- removing a menu item via the inspector moves focus to document's `body`

**Comment by draganescu** [comment]
More investigation resulted in doubting the origina bug:

- removing any block moves focus to the editor, removing a menu item via the canvas moves focus to the editor, moves focus to the editor `iframe`, but using `contentWindow.document.activeElement` shows the correct focus is applied to best available blocks in canvas
- removing a menu item via the inspector moves the focus properly can't repro consistently my finding above
- **removing a submenu items, both in canvas and in the inspector, this is where focus is lost.**

We need to manage focus where we do the transforms between link and submenu wither in the store like we foce select of previous block or in the blocks themselves.

**Comment by draganescu** [comment]
It seems there is a problem with `replaceBlocks` and how we use it to "transform" submenu to link. But many block operations use a `selectionHelper` that handles most selection following updating the block tree, including `REPLACE_BLOCKS`. Debugging this I learned that there are moments when the selection state is empty and because of the code [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/store/reducer.js#L1312) we return an empty selection state. Attempted a fix for this in #48105

The problem in this particular issue is a problem visible in the left hand list view as well, when replacing blocks the focus is not managed well and ends up on the `body` of the document.

**Comment by getdave** [comment]
@draganescu Great research. Thank you. As https://github.com/WordPress/gutenberg/pull/48105 was closed what are the next steps here?

**Comment by draganescu** [comment]
@getdave https://github.com/WordPress/gutenberg/pull/48204 is the solution. Waiting for it to be reviewed and to land asap.

**Comment by getdave** [comment]
Great work on https://github.com/WordPress/gutenberg/pull/48204 @draganescu 👏
